examples/lecture_materials/3_Metrics.md

```diff
@@ -0,0 +1,145 @@
+---
+jupyter:
+  jupytext:
+    text_representation:
+      extension: .md
+      format_name: markdown
+      format_version: '1.2'
+      jupytext_version: 1.5.0
+  kernelspec:
+    display_name: Python [conda env:.conda-raains]
+    language: python
+    name: conda-env-.conda-raains-py
+---
+
+# Performance Metrics
+
+We need to define reliable metrics to measure the quality of our model's detections.
+This entails two things:
+1. Assessing its ability to detect all of the objects that are present (no more, and no less), along with how well it localized those objects (i.e. how closely the predicted boxes match the true boxes)
+2. Assessing how consistently the detected objects are classified correctly.
+
+Both of these qualities can be assessed if we take our model's detections (post-NMS) and construct a so-called **confusion matrix** that compares the true objects to our predicted objects.
+
+## Confusion Matrix
+
+A confusion matrix displays all of the combinations of true/predicted categories that can manifest between our data's labels and our model's predictions.
+Consider the following figure, which displays the confusion matrix associated with the truth/predictions for a single image.
+
+
+<div style="text-align: center">
+<p>
+<img src="./pics/confusion_matrix.png" alt="An example of a confusion matrix associated with our model's predictions">
+</p>
+</div>
+
+
+This matrix essentially serves as a lookup table by which we can see how often the truth (row) aligns with a prediction (column).
+For example, the entry at `(row-rectangle, col-triangle)` tells us how many true rectangles were (mis)classified as triangles.
+Thus high tallies along the diagonals indicate good agreement between the truth and predictions, whereas off-diagonal elements convey misclassifications.
+
+The confusion matrix affords us information that allows us to both assess the quality of our model's detections as well as the quality of the model's classifications.
+Let's appreciate the diverse information conveyed by this matrix.
+
+
+<div style="text-align: center">
+<p>
+<img src="./pics/tp_tn_matrix.png" alt="Demonstrating the effect of NMS thresholding" width=600>
+</p>
+</div>
+
+
+### Detection Metrics: Precision & Recall
```

Comment:
totally optional but it might be worth covering mAP if you want to make this more detailed/complete